benchmark/src/submit_pattern.rs

```diff
@@ -0,0 +1,212 @@
+use logger::prelude::*;
+use structopt::StructOpt;
+
+/// Time interval to flood all TXNs.
+/// If flood takes less than this duration, we pause for the rest.
+const FLOOD_SUBMISSION_INTERVAL_MS: usize = 100;
+
+/// Control how TXNs are submitted to AC.
+#[derive(Debug, StructOpt, Clone, Copy)]
+#[structopt(name = "pattern", about = "")]
+pub enum SubmitPattern {
+    /// Submit all TXN requests in one shot without any control.
+    #[structopt(name = "flood_all")]
+    FloodAll(FloodAll),
+    /// Submit TXN requests at constant rate: #TXNs submitted per second = constant.
+    #[structopt(name = "const")]
+    ConstRate(ConstRate),
+    /// Submit TXN requests at linear speed: #TXNs submitted per second = slope * idx + intercept.
+    #[structopt(name = "linear")]
+    LinearRate(LinearRate),
+}
+
+#[derive(Debug, StructOpt, Clone, Copy)]
+pub struct FloodAll {}
+
+#[derive(Debug, StructOpt, Clone, Copy)]
+pub struct ConstRate {
+    /// Constant number of TXN requests submitted per second.
+    #[structopt(short = "c", long = "constant", default_value = "10")]
+    pub constant: usize,
+    /// Note about internal fields: structopt at master supports '#[structopt(skip)]`
+    /// but not in its latest release version yet. TODO: use `#[structopt(skip)]` when released.
+    /// Submission interval in milliseconds. Used internally, not to be set by user.
+    #[structopt(short = "i", long = "interval_ms", default_value = "1000")]
+    pub interval_ms: usize,
+}
+
+#[derive(Debug, StructOpt, Clone, Copy)]
+pub struct LinearRate {
+    /// Linear function's slope, e.g., increase X TXNs per second.
+    #[structopt(short = "a", long = "slope", default_value = "1")]
+    pub slope: usize,
+    /// Linear function's intercept, e.g., start with X TXNs per second. Must be integer.
+    #[structopt(short = "b", long = "intercept", default_value = "0")]
+    pub intercept: usize,
+    /// Linear function's initial index value. Used internally, not to be set by user.
+    #[structopt(short = "i", long = "interval_ms", default_value = "0")]
+    pub idx: usize,
+    /// Submission interval in milliseconds. Used internally, not to be set by user.
+    #[structopt(short = "i", long = "interval_ms", default_value = "1000")]
+    pub interval_ms: usize,
+}
```

Comment:
`StructOpt` is used for user facing CLI options and for application logic, and ending up with 
> Used internally, not to be set by user. 

Not great.

It'll be better to use `StructOpt` for the options needed from the user, and based on them, construct the structs needed in the app.

Reply:
After changes, in CLI we have `flood` and `const -c` sub-command to specify submit pattern. While internally, we use `ConstantRate` struct to combine both them.